#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.4955742664623031 0.7737268214671993 0.39465548540798817 1.5840753907875558
  position 1.0293662663834402 1.22499499463466 0.5855154260296379
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 3.5 3.5
}
Robot {
  translation 0 0.08 0
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Box {
        size 0.15 0.1 0.15
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.085 -0.03 0.04
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.0849999999996426 -0.029991128757063 0.039999999670549805
        rotation 2.903732224161803e-13 -2.904166521226548e-13 1 1.5707996938938997
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 1 0.886275 0.235294
              roughness 1
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.085 -0.03 -0.04
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.0849999999996328 -0.029991192024344946 -0.04000000033570439
        rotation 7.866201471360325e-13 -7.866649085088581e-13 1 1.5707996938938928
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 1 0.886275 0.235294
              roughness 1
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.085 -0.03 0.04
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.0850000000003574 -0.02999112889090875 0.03999999967054489
        rotation 2.9188435722925013e-13 -2.919270993202906e-13 1 1.5707996938938962
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 1 0.886275 0.235294
              roughness 1
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.085 -0.03 -0.04
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.08500000000036709 -0.02999119215819074 -0.04000000033570925
        rotation 7.892443193506957e-13 -7.892876863345107e-13 1 1.570799693893881
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 1 0.886275 0.235294
              roughness 1
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(4)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.04 -0.02 0.08
      rotation 0 1 0 0.187
      children [
        DEF SENSOR Shape {
          appearance PBRAppearance {
            baseColor 1 0.666667 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
    DistanceSensor {
      translation -0.04 -0.02 0.08
      rotation 0 1 0 -0.187
      children [
        DEF SENSOR Shape {
          appearance PBRAppearance {
            baseColor 1 0.666667 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    LightSensor {
      translation -0.04 0.02 0.08
      rotation 0 1 0 -0.1308996938995747
      children [
        USE SENSOR
      ]
      name "ls_right"
    }
    LightSensor {
      translation 0.04 0.02 0.08
      rotation 0 1 0 0.1308996938995747
      children [
        USE SENSOR
      ]
      name "ls_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "new_robot_avoid"
}
